gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java

```diff
@@ -652,21 +660,77 @@ public boolean exists() throws ComputeException {
    *     not exist, {@code false} if the state is not {@link Operation.Status#DONE}
    * @throws ComputeException upon failure
    */
-  public boolean isDone() throws ComputeException {
+  public boolean isDone() {
     Operation operation = compute.getOperation(operationId,
         OperationOption.fields(Compute.OperationField.STATUS));
     return operation == null || operation.status() == Status.DONE;
   }
 
   /**
+   * Blocks until this operation completes its execution, either failing or succeeding. This method
+   * returns current operation's latest information. If the operation no longer exists, this method
+   * returns {@code null}. By default, the operation status is checked every 500 milliseconds, to
+   * configure this value use {@link WaitForOption#checkEvery(long, TimeUnit)}. Use
+   * {@link WaitForOption#timeout(long, TimeUnit)} to set the maximum time to wait.
+   *
+   * <p>Example usage of {@code waitFor()}:
+   * <pre> {@code
+   * Operation completedOperation = operation.waitFor();
+   * if (completedOperation == null) {
+   *   // operation no longer exists
+   * } else if (completedOperation.errors() != null) {
+   *   // operation failed, handle error
+   * } else {
+   *   // operation completed successfully
+   * }}</pre>
+   *
+   * <p>Example usage of {@code waitFor()} with checking period and timeout:
+   * <pre> {@code
+   * Operation completedOperation =
+   *     operation.waitFor(WaitForOption.checkEvery(1, TimeUnit.SECONDS),
+   *         WaitForOption.timeout(60, TimeUnit.SECONDS));
+   * if (completedOperation == null) {
+   *   // operation no longer exists
+   * } else if (completedOperation.errors() != null) {
+   *   // operation failed, handle error
+   * } else {
+   *   // operation completed successfully
+   * }}</pre>
+   *
+   * @param waitOptions options to configure checking period and timeout
+   * @throws ComputeException upon failure
+   * @throws InterruptedException if the current thread gets interrupted while waiting for the
+   *     operation to complete
+   * @throws TimeoutException if the timeout provided with
+   *     {@link WaitForOption#timeout(long, TimeUnit)} is exceeded. If no such option is provided
+   *     this exception is never thrown.
+   */
+  public Operation waitFor(WaitForOption... waitOptions)
+      throws InterruptedException, TimeoutException {
+    Map<WaitForOption.Option, ?> optionMap = WaitForOption.asMap(waitOptions);
```

Comment:
I like the change! and fine as is.

One suggestion though.

Because you made `WaitForOption` public (for making it common) you also exposed many elements that otherwise would be package scope.
I suggest to minimize those as much as possible. One way that may look less "internal" would be to eliminate WaitForOption.Option and its related methods and instead make `CheckingPeriod` and Timeout (a new inner class) sub-classes of `WaitForOption`. These 2 classes can have a public static `getOrDefault(WaitForOption... options)` which can use an internal/private enum to avoid "instanceof" and have specific methods to get their values.